### ISSUE TITLE
Error: Deserializing a `typing.Sequence`

### ISSUE THREAD

**Issue by stdedos** [rnag/dataclass-wizard#156]
* Dataclass Wizard version: 0.26.1 / 0.30.1
* Python version: 3.9
* Operating System:

### Description

Describe what you were trying to get done.
Tell us what happened, what went wrong, and what you expected to happen.

I am trying to load existing configuration into the following definition:

```py
@dataclass
class BaseOptions(JSONSerializable):
    """Base class for configuration options objects."""

    class Meta(JSONSerializable.Meta):
        key_transform_with_dump = "SNAKE"
        raise_on_unknown_json_key = True
        skip_defaults = True

    # def __getitem__(self, item):
    #     return getattr(self, item)
    #
    # def __setitem__(self, key, item):
    #     return setattr(self, key, item)


@dataclass
class Options(BaseOptions):
    """Configuration options object for glira."""

    email: str = ""
    token: str = ""
    fields: Sequence[IssueFields] = (
        Fields.A,
        Fields.B,
        Fields.C,
        IssueFields.D,
        IssueFields.E,
    )
```

However, since my current configuration has no `fields:`, it fails with

### What I Did

```console
$ ./app.py
INFO:dataclass_wizard:DEBUG Mode is enabled
Traceback (most recent call last):
  File "p/.venv/lib/python3.9/site-packages/dataclass_wizard/loaders.py", line 567, in fromdict
    load = CLASS_TO_LOAD_FUNC[cls]
KeyError: <class 'r.w.s.s.j.c.b.Options'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  ...
  File "p/src/r/w/s/s/j/c/a.py", line 42, in load
    return self.configuration_class.from_dict(config_data)
  File "p/.venv/lib/python3.9/site-packages/dataclass_wizard/loaders.py", line 569, in fromdict
    load = load_func_for_dataclass(cls)
  File "p/.venv/lib/python3.9/site-packages/dataclass_wizard/loaders.py", line 623, in load_func_for_dataclass
    field_to_parser = dataclass_field_to_load_parser(cls_loader, cls, config)
  File "p/.venv/lib/python3.9/site-packages/dataclass_wizard/class_helper.py", line 121, in dataclass_field_to_load_parser
    return _setup_load_config_for_cls(cls_loader, cls, config, save)
  File "p/.venv/lib/python3.9/site-packages/dataclass_wizard/class_helper.py", line 198, in _setup_load_config_for_cls
    name_to_parser[f.name] = cls_loader.get_parser_for_annotation(
  File "p/.venv/lib/python3.9/site-packages/dataclass_wizard/loaders.py", line 379, in get_parser_for_annotation
    return OptionalParser(
  File "<string>", line 4, in __init__
  File "p/.venv/lib/python3.9/site-packages/dataclass_wizard/parsers.py", line 204, in __post_init__
    self.parser: AbstractParser = get_parser(self.base_type, cls, extras)
  File "p/.venv/lib/python3.9/site-packages/dataclass_wizard/loaders.py", line 471, in get_parser_for_annotation
    raise ParseError(
dataclass_wizard.errors.ParseError: Failure parsing field `None` in class `None`. Expected a type Sequence, got NoneType.
  value: None
  error: Provided type is not currently supported.
  unsupported_type: <class 'collections.abc.Sequence'>
```

I also tried setting `Optional[Sequence[IssueFields]]`; no change.

I tried perusing https://dataclass-wizard.readthedocs.io/ but I didn't pinpoint my situation (a `@dataclass` that contains a default value for a config that does not when loaded)

**Comment by rnag** [comment]
Hi @stdedos, thanks for opening this issue! A friendly reminder to include a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) (MRE) when opening an issue, as it helps in troubleshooting more effectively.

After investigating, it seems the issue arises because the library does not currently support the `Sequence` type. As a workaround for now, you can use:
- `list[IssueFields]`
- `tuple[IssueFields, ...]`

I'm planning to introduce support for `Sequence` soon, so stay tuned!

**Comment by rnag** [comment]
@stdedos I forgot to link the PR, but this should be resolved as part of `v0.32`, which adds support for ABC Containers `Sequence`, `MutableSequence`, and `Collection`.

Working example:

```python3
from typing import Sequence
from dataclasses import dataclass

from dataclass_wizard import JSONWizard


@dataclass
class IssueFields:
    name: str


@dataclass
class Options(JSONWizard):
    email: str = ""
    token: str = ""
    fields: Sequence[IssueFields] = (
        IssueFields('A'),
        IssueFields('B'),
        IssueFields('C'),
    )

opt = Options.from_dict({
    'email': 'a@b.org',
    'token': '<PASSWORD>',
    # 'fields': [{'Name': 'X'}, {'Name': 'Y'}]
})

print(repr(opt))
#> Options(email='a@b.org', token='<PASSWORD>', fields=(IssueFields(name='A'), IssueFields(name='B'), IssueFields(name='C')))
```

**Comment by stdedos** [comment]
> A friendly reminder to include a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) (MRE) when opening an issue, as it helps in troubleshooting more effectively.

Apologies; I thought the classes were enough 😓 

But thank you for figuring it out and fixing it 🎉
